File: C/real world problem solve/to_do_list.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 100

struct ToDoList {
    char tasks[MAX_SIZE][100];
    int size;
};

void initList(struct ToDoList* list) {
    list->size = 0;
}

void addTask(struct ToDoList* list, char task[]) {
    if (list->size == MAX_SIZE) {
        printf("List is full! Cannot add more tasks.\n");
        return;
    }
    
    strcpy(list->tasks[list->size], task);
    list->size++;
}

void displayTasks(struct ToDoList* list) {
    if (list->size == 0) {
        printf("No tasks in the list.\n");
        return;
    }
    
    printf("Tasks:\n");
    for (int i = 0; i < list->size; i++) {
        printf("%d. %s\n", i + 1, list->tasks[i]);
    }
}

int main() {
    struct ToDoList myToDoList;
    initList(&myToDoList);
    
    addTask(&myToDoList, "Buy groceries");
    addTask(&myToDoList, "Pay bills");
    addTask(&myToDoList, "Do laundry");
    
    displayTasks(&myToDoList);

    return 0;
}

File: Java/examples/TodoList.java
import java.util.ArrayList;

public class TodoList {
    public static int MAX_SIZE = 100;
    public static ArrayList<Todo> tasks = new ArrayList<>();

    public static void main(String[] args) {
        initList();

        addTask("Buy groceries", false);
        addTask("Pay bills", true);
        addTask("Do laundry", false);

        displayTasks();
    }

    public static void initList() {
        tasks.clear();
    }

    public static void addTask(String task, boolean done) {
        if (tasks.size() == MAX_SIZE) {
            System.out.println("List is full, cannot add more tasks!");
            return;
        }
        
        Todo todo = new Todo(task);

        todo.setDone(done);

        tasks.add(todo);
    }

    public static void displayTasks() {
        if (tasks.size() == 0) {
            System.out.println("No tasks in the list!");
            return;
        }

        System.out.println("=====Tasks=====");

        for(Todo task : tasks) {
            System.out.println("Task: " + task.task + " | Done: " + task.done);
        }
    }
}


class Todo {
    public String task;
    public boolean done = false;

    public Todo(String name) {
        this.task = name;
    }

    public void setDone(boolean newStatus) {
        done = newStatus;
    }
}